documentation/release_notes.markdown

```diff
@@ -7,24 +7,311 @@ canonical: "/puppetdb/latest/release_notes.html"
 3.0.0
 -----
 
-### Upgrading
+### Before Upgrading
+
+* We have retired support for versions of PostgreSQL prior top 9.3. If you are using HSQLDB for production,
+we also recommended switching to PostgreSQL at least 9.3, as HSQLDB has a number of scaling
+and operational issues and is only recommended for testing and proof of concept installations.
+
+* Wwe advise that users install the PostgreSQL extension `pg_trgm` for increased
+indexing performance for regular expression queries. Using the command `create extension pg_trgm`
+as PostgreSQL super-user and before starting PuppetDB will allow these new indexes to be created.
+
+* Ensure during a package upgrade that you analyze any changed configuration files. For Debian
+you will receive warnings when upgrading interactively about these files, and for RedHat based
+distributions you will find that the RPM drops .rpmnew files that you should diff and ensure
+that any new content is merged into your existing configuration.
+
+* Make sure all your PuppetDB instances are shut down and only upgrade one at a time.
+
+* As usual, don't forget to also upgrade your puppetdb-terminus package
+(on the host where your Puppet Master lives), and restart your
+master service.
+
+* If you receive the error "Could not open
```

Comment:
We had dropped this point and the next from the 2.3.0 release notes, as it seemed redundant, that is we haven't seen it for quite some time, and a google search will find this solution in this document anyway. That is, no need to keep saying it.